### PR TITLE
Adding documentation and specs

### PR DESCRIPTION
**Pull request by jeremyf**
## Clarifying Hyrax::Operation

@7d793190752e63c32310b43186c2cddd598c3b17

Adding documentation and additional specs to better understand these
classes.

## Renaming parameter to reflect its implementation

@c3c811fd9dbb079f601ecb9d68f9f4731204f35e

Given that we have a class named Operation but the variable was "log",
rename the variable to "operation" to better reflect its usage.

@projecthydra-labs/hyrax-code-reviewers

### CODE REVIEW

File: spec/models/hyrax/operation_spec.rb [hunk 1]
```diff
@@ -12,6 +12,19 @@
       end
     end
 
+    describe "with only failng processes" do
```
Comment (jcoyne) [inline spec/models/hyrax/operation_spec.rb]: typo here.

File: app/models/hyrax/operation.rb [hunk 2]
```diff
@@ -11,33 +12,57 @@ class Operation < ActiveRecord::Base
...
 
     # If this is a batch job (has children), check to see if all the children are complete
+    # @see #fail! when none of the children are in a PENDING nor PERFORMING state but at least one is in a FAILURE state
+    # @see #success! none of the children are in a PENDING, PERFORMING, nor FAILURE state
     def rollup_status
       with_lock do
-        stats = children.pluck(:status)
+        # We don't need all of the status of the children, just need to see
+        # if there is at least one PENDING, PERFORMING, or FAILURE.
+        stats = children.select(:status).distinct.pluck(:status)
```
Comment (mjgiarlo) [inline app/models/hyrax/operation.rb]: Can you say more about why this change is needed?

File: spec/models/hyrax/operation_spec.rb (hunk 1 above)
Comment (mjgiarlo) [inline spec/models/hyrax/operation_spec.rb]: I'd also like to see `s/describe/context/` here, but it's no big.

File: app/models/hyrax/operation.rb (hunk 2 above)
Comment (jeremyf) [inline app/models/hyrax/operation.rb]: Added comments inline.

The gist is: instead of transferring all of the statuses of the children from the database, to Rails, and then instantiating an array and leveraging the include that checks the whole array, opt instead to rely on the database to minimize the return values to that which is needed.

**Comment by jeremyf** [comment]
Added comments inline.

The gist is: instead of transferring all of the statuses of the children from the database, to Rails, and then instantiating an array and leveraging the include that checks the whole array, opt instead to rely on the database to minimize the return values to that which is needed.
